kogito-codegen-modules/kogito-codegen-processes/src/main/java/org/kie/kogito/codegen/process/ProcessCodegen.java

```
@@ -582,4 +595,50 @@ public Optional<ApplicationSection> section() {
     public int priority() {
         return 10;
     }
+
+    private void processProducerTemplate(Optional<Expression> statement) {
+        TemplatedGenerator generator = TemplatedGenerator.builder()
+                .withTemplateBasePath("/class-templates/producer/")
+                .withFallbackContext(JavaKogitoBuildContext.CONTEXT_NAME)
+                .withTargetTypeName(BUSINESS_CALENDAR_PRODUCER_TEMPLATE)
+                .build(context(), BUSINESS_CALENDAR_PRODUCER_TEMPLATE);
+        CompilationUnit compilationUnit = generator.compilationUnitOrThrow();
+
+        statement.ifPresent(expression -> compilationUnit.findFirst(ClassOrInterfaceDeclaration.class)
+                .flatMap(NodeWithMembers::getDefaultConstructor)
+                .ifPresent(constructor -> constructor.getBody().getStatements().stream()
```

Here the code should throw an exception, because the DefaultConstructor must be present in the template